app/views/dashboard/advisers/_advisers_table.html.erb

```diff
@@ -9,18 +9,29 @@
       </tr>
     </thead>
     <tbody>
-      <% advisers.each do |adviser| %>
-        <tr class="t-advisers-table-row">
-          <td class="t-adviser-reference-number"><%= adviser.reference_number %></td>
-          <td class="t-adviser-name"><%= adviser.name %></td>
-          <td>
-            <ul class="l-comma-separated-list">
-              <% adviser.qualifications.each do |qualification| %>
-                <li class="t-adviser-qualification"><%= qualification.name %></li>
-              <% end %>
-            </ul>
+      <% if firm.advisers.present? %>
+        <% firm.advisers.each do |adviser| %>
+          <tr class="t-advisers-table-row">
+            <td class="t-adviser-reference-number"><%= adviser.reference_number %></td>
+            <td class="t-adviser-name"><%= adviser.name %></td>
+            <td>
+              <ul class="l-comma-separated-list">
+                <% adviser.qualifications.each do |qualification| %>
+                  <li class="t-adviser-qualification"><%= qualification.name %></li>
+                <% end %>
+              </ul>
+            </td>
+            <td class="t-adviser-postcode"><%= adviser.postcode %></td>
+          </tr>
+        <% end %>
+      <% else %>
+        <tr>
+          <td class="t-no-advisers-message" colspan="4">
+            <%= t('dashboard.advisers_index.no_advisers_message') %>
+            <%= link_to(t('dashboard.advisers_index.add_adviser_button_full', firm_name: firm.registered_name),
+                        new_dashboard_firm_adviser_path(firm),
+                        class: 't-add-adviser-link')  %>.
```

Comment:
Strictly we would want to duplicate it. It's effectively non-internationalised text, and it's reusing a key that doesn't apply to it. This isn't an `add_adviser_button_full`, it's an `add_adviser_link` — that semantic distinction means we have to add the full stop, therefore enough of a semantic distinction to mean it needs another key. 

Imagine you didn't know what the text really was — would you feel confident using the key here?

That said, who cares. It'd be enough to make it explicit with a `+ '.'` in the code, just so it's obvious it's meant to be there.